### DISCUSSION TITLE
Postgres Full Text Search

### DISCUSSION THREAD

**Discussion by ivank** [directus/directus#24034] [Feature Requests]
### Summary

Implement postgres' [full text search](https://www.postgresql.org/docs/current/textsearch.html).

### Motivation

You cannot index fields bigger than 2704 chars, which is laughably small for any blog-like environment.

We are seeing errors like:
```
ERROR: insert into "block_richtext" ("content", "date_created", "user_created") values ($1, $2, $3) returning "id" - index row size 3968 exceeds btree version 4 maximum 2704 for index "block_richtext_content_index"
```

### Detailed Design

When a field is marked as "indexed" and the field type is "text" switch from using normal b-tree based index in postgres to full text search, and use ts_vector for it.

### Must Have:
- Index search for text fields of bigger size

### Drawbacks

It would mean redoing some search queries, but would ultimately add more flexability in using directus search _directly_ by users

### Alternatives

Elastic search implementation

### Adoption Strategy

.

**Comment by JayShoe** [comment]
I’m rallying support for a feature called `facets`, which could solve for this issue by centralizing and abstracting special search functions into reusable and extensible schema definitions. Facets work because they:

- **Allow tailored search behaviors**: Define full-text search in Postgres, range filtering, geolocation queries, and more in a reusable way.  
- **Support multiple search definitions per collection**: Enable targeting of specific fields for specific purposes, independent of a single database's limitations.  
- **Enable advanced integrations**: Seamlessly translate into queries for tools like Elasticsearch or Algolia, simplifying scalable and extensible search functionality.

Check it out: [[Facets Proposal](https://github.com/directus/directus/discussions/23605)](https://github.com/directus/directus/discussions/23605)
